Source/WebCore/css/parser/CSSPropertyParserConsumer+Animations.cpp

```diff
@@ -42,8 +42,9 @@ RefPtr<CSSValue> consumeKeyframesName(CSSParserTokenRange& range, const CSSParse
 
     if (range.peek().type() == StringToken) {
         auto& token = range.consumeIncludingWhitespace();
-        if (equalLettersIgnoringASCIICase(token.value(), "none"_s))
-            return CSSPrimitiveValue::create(CSSValueNone);
+        auto valueId = cssValueKeywordID(token.value());
+        if (isValidCustomIdentifier(valueId) && valueId != CSSValueNone)
```

Comment:
Would you mind adding a comment explaining that there is prose indicating that "none" must be excluded. Somethings like:

```c++
// Per spec, "The <custom-ident> additionally excludes the `none` keyword."
```

Not necessary for this change, but perhaps adding a `isValidCustomIdentifierExcluding<CSValueID... IdsToExclude>(CSSValueID)` would be a good helper in CSSParserIdioms.h.

I have been meaning to suggest to the CSSWG, and this reminds me I should go do that, an extension to the grammar syntax to allow additional exclusions. Something along the lines of <custom-ident excluding="none, foo"> to make these rules more clear.